### PR TITLE
Download Grafana Agent on FreeBSD

### PR DESCRIPTION
**Pull request by patmaddox**
### Change description

This change adds FreeBSD support by automatically downloading the grafana agent FreeBSD build.

### Additional details and screenshots

GitHub actions doesn't support FreeBSD, but [Cirrus CI does](https://cirrus-ci.org/guide/FreeBSD/). I also added a Cirrus CI config for FreeBSD.

### Checklist

I don't have an example application to share, but we have used this branch to push data and graphs to AWS managed Prometheus and Grafana (it worked great!)

- [x] I have added unit tests to cover my changes.
- [x] I have added documentation to cover my changes.
- [x] My changes have passed unit tests and have been tested E2E in an example project.

### CODE REVIEW

File: .cirrus.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+freebsd_instance:
```
Comment (akoutmos) [inline .cirrus.yml:1]: I am not familiar with cirrus....so perhaps this is a silly question. Why is this file required since the freebsd build is already on the GrafanaAgent release artifact list?

Comment (patmaddox) [reply .cirrus.yml:1]: It's to run the test suite on FreeBSD so that:

1) The new config that downloads the agent actually gets exercised
2) We have confidence that the test suite passes on a FreeBSD machine

I'm not opposed to leaving it out - I don't see anything in PromEx that would have issues on FreeBSD. It's just a safety net so that you don't release a version that doesn't work on FreeBSD. Or at least we would know that a version is incompatible with FreeBSD.

There is actually one scenario where I can see it being relevant: the FreeBSD ports for [erlang](https://cgit.freebsd.org/ports/tree/lang/erlang/Makefile) and [elixir](https://cgit.freebsd.org/ports/tree/lang/elixir/Makefile) usually take from a few days to a few weeks to be updated after there's a new release. So if PromEx used a feature in a brand new Elixir release (e.g. `PartitionSupervisor` in 1.14) then the test suite would fail on FreeBSD until the port is updated to the new Elixir version.

Perhaps it's possible to make Cirrus optional, so that a failed build there isn't considered a project-wide build failure? Instead it could just report on the state of current PromEx with whatever version of Erlang/Elixir are on FreeBSD?

Comment (patmaddox) [reply .cirrus.yml:1]: I guess if a dep used a feature unavailable in the current Elixir, it would just fail to compile, right? So the including project's CI would fail, and I'd know to pin `prom_ex` to a compatible version.

Anyway, I included the Cirrus config because it was easy to set up, and just to cover the bases.

But I think it's reasonable to remove it, and just have this PR be about the agent checksums.

How does that sound?

Comment (akoutmos) [reply .cirrus.yml:1]: Sorry for the delay. Been a crazy work day. Let's remove Cirrus for now and just add support for FreeBSD and the checksums as you mentioned. Thanks!

Comment (patmaddox) [reply .cirrus.yml]: Done!

Comment (akoutmos) [reply .cirrus.yml]: Awesome! Thanks!

**Review by akoutmos — APPROVED** [review]
LGTM! Thanks for adding FreeBSD support :tada:
